src/d3a/models/strategy/external_strategy.py

```diff
@@ -0,0 +1,125 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+from d3a.models.strategy import BaseStrategy
+
+from redis import StrictRedis
+import json
+from d3a.d3a_core.redis_communication import REDIS_URL
+from d3a.models.market.market_structures import trade_bid_info_from_JSON_string
+
+
+class RedisExternalConnection:
```

Comment:
Note, need to merge this class with the MarketRedisAPI class.

Reply:
Ahhhh, ok. So we agree. This is the answer to my comment here   https://github.com/gridsingularity/d3a/pull/567/files#r346193322  right?

Reply:
Are you planning to move this as mentioned or only after tomorrow?

Reply:
Actually, let me do it now, better to have it in the first place than forgetting it later.